test/execution_plan_hooks_test.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'test_helper'
+
+module Dynflow
+  class ExecutionPlan
+    describe Hooks do
+      include PlanAssertions
+
+      let(:world) { WorldFactory.create_world }
+
+      class Flag
+        class << self
+          def raise!
+            @raised = true
+          end
+
+          def raised?
+            @raised
+          end
+
+          def lower!
+            @raised = false
+          end
+        end
+      end
+
+      class FlagHook < ::Dynflow::ExecutionPlan::Hooks::Abstract
+        def on_success(_execution_plan, _action)
+          Flag.raise!
+        end
+
+        def on_stopped(_execution_plan, _action)
+          Flag.raise!
+          raise "A controlled failure"
+        end
+      end
+
+      class ActionWithHooks < ::Dynflow::Action
+        execution_plan_hooks.use FlagHook, :on => :success
```

Comment:
I wonder if we need a class here vs. just action methods. Seems a bit strange that I need to define both `on_success` in the hook and `:on => :success` here. What about something like this:

```ruby
execution_plan_hooks.run :send_notification, :on => :stopped
```

Reply:
And then the shareable parts would be modules which would provide the method and which would register itself when included?

Reply:
Yes, alternatively, we could still stay with the hook classes, but then, I would recommend limiting it to define the `execute` methods, and not using the `on_event`-like method names. It was confusing for me to see the `:on => success` when registering the hook, and seeing also `on_success` in the hook itself.